crates/rpc/tests/rpc.rs

```diff
@@ -0,0 +1,488 @@
+        .await
+        .expect("getClass failed");
+}
+
+#[tokio::test]
+async fn test_get_class_hash_at() {
+    let TestContext {
+        client,
+        block: _,
+        block_id,
+        extracted_value: deploy_tx_hash,
+    } = context(|block| {
+        block.transactions.iter().find_map(|transaction| match transaction {
+            Transaction::DeployAccount(DeployAccountTransaction::V3(
+                deploy,
+            )) => Some(deploy.transaction_hash),
+            _ => None,
+        })
+    })
+    .await;
+
+    let receipt = match client
+        .get_transaction_receipt(deploy_tx_hash)
+        .await
+        .expect("the transaction to have a matching receipt")
+    {
+        MaybePendingTransactionReceipt::Receipt(
+            TransactionReceipt::DeployAccount(receipt),
+        ) => receipt,
+        _ => panic!("Expected a valid receipt, got a pending one"),
+    };
+
+    client
+        .get_class_hash_at(block_id, receipt.contract_address)
+        .await
+        .expect("getClass failed");
+}
+
+#[tokio::test]
+async fn test_get_nonce() {
+    let TestContext {
+        client,
+        block: _,
+        block_id,
+        extracted_value: (original_nonce, sender),
+    } = context(|block| {
+        // Browse the transaction in reverse order to make sure we got the latest nonce.
+        block.transactions.iter().rev().find_map(
+            |transaction| match transaction {
+                Transaction::Invoke(InvokeTransaction::V1(invoke)) => {
+                    Some((invoke.nonce, invoke.sender_address))
+                }
+                Transaction::Invoke(InvokeTransaction::V3(invoke)) => {
+                    Some((invoke.nonce, invoke.sender_address))
+                }
+                _ => None,
+            },
+        )
+    })
+    .await;
+    let original_nonce = truncate_felt_to_u128(&original_nonce);
+
+    let incremented_nonce =
+        client.get_nonce(block_id, sender).await.expect("getNonce failed");
+    let incremented_nonce = truncate_felt_to_u128(&incremented_nonce);
+
+    assert_eq!(original_nonce + 1, incremented_nonce);
+}
+
+#[tokio::test]
+async fn test_get_transaction_by_block_id_and_index() {
+    let TestContext { client, block, block_id, extracted_value: () } =
+        latest_10_tx_context().await;
+
+    async fn check_transaction(
+        client: &JsonRpcClient<HttpTransport>,
+        block_id: BlockId,
+        transaction_index: usize,
+        expected: &BlockWithTxs,
+    ) {
+        let transaction = client.get_transaction_by_block_id_and_index(block_id, transaction_index as u64).await.expect("Failed to retrieve a specific transaction by its block ID and index");
+
+        // `starknet-rs` doesn't implement `PartialEq` on its DTOs, and transactions have many *variants which makes pure Rust comparison painful.
+        // Just serialize these to compare them.
+        assert_eq!(
+            serde_json::to_string(&transaction).unwrap(),
```

Comment:
no need to `serde_json::to_string`, as the `serde_json::Value` is `Eq` and also can deal with fields ordering

we might also want to use [`pretty_assertions`](https://docs.rs/pretty_assertions/latest/pretty_assertions/) for complex values for better readability in case test fails

Reply:
Ah! The choice for `to_string` was a poor man's `pretty_assertions` :) Thanks!